### ISSUE TITLE
Sync not work on ios

### ISSUE THREAD

**Issue by Porco24** [vrtmrz/obsidian-livesync#204]
I use this plugin a long time, and it's always work on pc
About two to three months ago, on my iPad, Obsidian would crash upon opening as long as this plugin was installed.
Additionally, about half a month ago, I encountered the same issue on my iPhone. So, I deleted the Obsidian app on my iPhone and reinstalled the app and the plugin. However, since then, abot half of MD files I cann't sync to the repository, and even using Fetch, I can't retrieve the full repository data anymore.
I haven't seen many logs, only the one in the screenshot. I'm not sure if it can provide any help.
![82cf8832eb0c1b7199f1bad241652b6](https://user-images.githubusercontent.com/25788636/234027783-c4577cda-cdd7-484f-a9bf-23305fb1e6d3.jpg)

**Comment by vrtmrz** [comment]
Sorry for your inconvenience and thank you so much for opening this issue.

Would you please check if the LiveSync data has been synchronised using `Pick a file to show history`?
If the data is simply synchronised but not realised in storage, some files will be shown in the list. 
This behaviour sometimes occurs, especially in mobile devices when the device has slept and the replication and fetching chunks are cancelled.

If some missing files are shown, I am sorry to have to ask you to do this, could you please try `Fetch` again with the vault that already synchronised half, and keep awake the device?

**Comment by Porco24** [comment]
Hi,I am sorry about I cant find the “ Pick a file to show history”, would u like give me some hint?

**Comment by Sirtz** [comment]
@Porco24 thats in the command palette

**Comment by Porco24** [comment]
@Sirtz Thanks your so much, I found it now

@vrtmrz 
I used "Pick a file to show history" and found that the files have not been synchronized.
And I tried to Fetch again, I get this error
![image](https://user-images.githubusercontent.com/25788636/235157158-76fa574c-5556-44b3-8fcf-4fbb20d58b3d.png)

**Comment by vrtmrz** [comment]
@Porco24 
Thank you for confirming this! Hmmm. Does the replication error also appear on other devices?

**Comment by Porco24** [comment]
@vrtmrz 
Hi,
I've never had any issues syncing between my home and office computers, but a few months ago, my iPad suddenly stopped updating without any error messages. Later, when I opened Obsidian on my iPad, the app would first display some sync plugin logs with content I can't remember, then show a loading progress bar before crashing. Recently, I tried uninstalling Obsidian from my iPad and syncing the repository again using the sync plugin. However, it encountered a problem similar to what I'm experiencing on my iPhone, where not all files have been synchronized. When I try to fetch, there are only logs saying "Hidden files and plugin synchronization have been temporarily disabled" and "All synchronization have been temporarily disabled", with no progress indication afterwards.

The issue I'm facing on my iPhone is quite similar to that on my iPad. If there were more log outputs, perhaps I could be of more assistance.

**Comment by vrtmrz** [comment]
Thank you for checking things!
I am wondering why it has crashed in this case. One of that had been caused by Out-Of-Memory on `Plugin and their settings`, And it has been refined as `Customization sync` at v0.19.0.
If you enabled this feature, would you mind if I ask you to try the latest version?

If it does not work well, possibly rebuilding is needed. 
May we ask if other devices have been synchronised?

**Comment by Porco24** [comment]
Yes, I had enable sync Plugin, and I had try Customization sync at v0.19.0. 
It's still have the "replication error" when I open this setting and fetch again

My other pc devices is alwasy been synchronised

**Comment by vrtmrz** [comment]
Thank you for the detail! I am still wondering why it has been broken only on iOS.
Could you please try enabling a verbose log? Some more information will be shown in the log pane.
And, if you do not mind, it would be helpful if you could paste the report which can be made in the `Hatch` pane.

**Comment by Porco24** [comment]
@vrtmrz 
When I replied earlier, I had actually already turned on the verbose log, and it's didn't have more infomation.
And, about paste the report, I've redacted some personal information and attached it in the file.
[log.txt](https://github.com/vrtmrz/obsidian-livesync/files/11369982/log.txt)

**Comment by vrtmrz** [comment]
@Porco24 Oh, my apologies for that! I was inadvertent.

Thank you for the configuration! I made the same configuration and try to initialise - replicate and fetch between an iOS and a Windows device. However, contrary to my expectations, it succeeded.
I hope to investigate it more. Let me ask something.
- Messages which you have seen mean that the plug-in tried to fetch. Following this, (I hope that) you should see `Local Database Reset`, `Mark this device as resolved`  and `Replication activated`. Have you seen them?
- If none of them you have seen, may the sequence have been hung up. Possibly putting `redflag3.md` on top of your vault and restarting obsidian may help us. (This means that reset the local database and fetching from the remote without UI).
- When finally not worked, If possible, could you please try another CouchDB instance without SSL configuration on it for testing? (I am using fly.io for testing).

Sorry and thank you for your patience.

**Comment by Porco24** [comment]
@Sirtz 
However, this time when I opened Obsidian, it showed the following log, and after fetching and crashing again at around 1500 downloads, it displayed the same error. Since this is the first time I've encountered this log, I'm not sure if it's related to the current issue.
![01683cda390a46c77320e465b475a2b](https://user-images.githubusercontent.com/25788636/235929670-c7a5c184-be91-4bb6-9324-9009bc1c97a2.jpg)

**Comment by Porco24** [comment]
Additionally, I think SSL should not be an issue, as on my server, there are two repositories in the same CouchDB instance used for syncing Obsidian, and the other repository works fine on both iOS and Windows.However, their sync plugin settings are different.

**Comment by Porco24** [comment]
Hello, I've conducted a few simple tests. At first, I thought it was because my app wasn't completely removed, so I thoroughly cleared all cache data and reinstalled Obsidian. However, when I tried to fetch server data using the "copy URL" method, the Obsidian crash issue I mentioned before reappeared. Obsidian displayed a black screen, then reloaded the app, and the synchronization was interrupted.
I suspect that after this interruption, some dirty data was generated, which led to the fetch failure issue I mentioned earlier.
However, I do not know what caused Obsidian to crash.

**Comment by vrtmrz** [comment]
Sorry for being absent for a few days! 
Your information helps me, possibly memory has run out.

I am sorry you have tried so many times, could you please try once again with `Read chunks online` enabled? It may reduce the number of chunks and documents waiting.
Possibly enlarging `Chunk size` works for this, however, it requires rebuilding.

**Comment by vrtmrz** [comment]
Thank you too for your patience!
We can see the log by the document box icon on our ribbon. 
However, conflicted files can be picked up by `Pick a file to resolve conflict`. It might be an easier way.

**Comment by Porco24** [comment]
Oh, Thank you, I didn't even notice this feature before. I should have submitted the log when reporting the issue earlier. I apologize for my oversight.

**Comment by Porco24** [comment]
Hi, I have some good news: the sync on iOS is finally working correctly. I will summarize below the simple tests I conducted and what happened during the process.

First, I tried enabling the "Read chunks online" feature and set the "Chunks Size" to 100. After this, I attempted to fetch, but my Obsidian still showed the same error as before.Even after adjusting the "Chunks Size" to 200, it still didn't work.

After rebuilding my database, I enabled "Read chunks online" on my iPhone and set the "Chunks Size" to 100, then tried to fetch again. This time, my Obsidian relaunched itself once, and then attempted to relaunch again. However, the second relaunch failed and the screen remained black. I closed Obsidian and reopened the app, then turned on "Livesync." Miraculously, it fetched all the data from the server, and since then, Obsidian has not relaunched again.

On my iPad, I also set up the chunks and fetched server data. However, the iPad didn't encounter any relaunch issues.

So, it seems that the rebuild worked, but I don't know why, and I don't know why my iOS devices suddenly encountered this issue during use and required a rebuild. As for my iPhone's relaunch and black screen issue, I think it might be due to insufficient memory, but I'm not entirely sure about the specifics.

In summary, thank you for your help. The issue on my iOS devices has been resolved, and this is a great plugin.I would like to buy you a coffee

**Comment by Porco24** [comment]
Additionally, I have another question. After using the chunks size feature, my Livesync has become very slow. If increasing the size can solve this issue, is there a recommended size?

**Comment by vrtmrz** [comment]
Thank you for your great patience and for trying so many things!

I still think that it has been caused by out-of-memory for some chunks.
Chunk size controls how binary files are split. If too finely divided, it takes longer to wait. Conversely, if they are split too large divided, it takes longer to collect them. Therefore, it may depend on the type and number of files. This is just my guess, but maybe it can be balanced at around 50.

May I ask which process has slowed down? (`Show status inside editor` will help us).
One more thing, once the vault has been synchronised, we can safely disable `Read chunks online` relative to initial synchronisation. Will this improve the situation?

However, possibly we need a safety relief valve for fetching and synchronisation.

**Comment by Porco24** [comment]
Yes, but I found that the issue is not with the chunk size. It seems that there is a slight problem with Livesync on iOS. When I enable Livesync on iOS, the documents I modify on my PC are not updated in real-time on my iOS devices. The synchronization only happens when I reopen Obsidian, but it could be due to some of my settings.

Regardless, the issue was resolved when I turned off Livesync and switched to another sync method. I appreciate your patience, and I will close this issue.
